### ISSUE TITLE
Train and evaluate first RE model for `(GENE, is-in, BRAIN_REGION)`

### ISSUE THREAD

**Issue by FrancescoCasalegno** [BlueBrain/Search#613]
## Context 
- In #606 we collected annotated sentences for the relation extraction (RE) task for the relation type `(GENE, is-in, BRAIN_REGION)`.
- Effectively, this can be seen as an example of sentence classification.
- This Issue is now about implementing and evaluating different RE model.

## Actions
- [ ] What is the percentage of sentences that the human expert discarded due to wrong NER annotations?
  Note that
- [ ] Out of the annotated samples, what is the percentage of samples that are labelled as "not is-in"?
- [ ] (Given the previous point:) What is the performance (precision, recall, f1-score) of a naïve RE model that predicts `is-in` anytime there's a co-mention of a `GENE` and a `BRAIN_REGION` in the same unit of text? 
- [ ] What is the performance of a pre-trained NLI model that takes as `premise` the paragraph and as `hypothesis` a sentence like `[gene] is located in [brain region]`?
- [ ] What is the performance of a pre-trained binary-QA model that takes as `context` the paragraph and as `hypothesis` a sentence like `Is [gene] located in [brain region]?`?
- [ ] What is the performance of a custom trained relation-extraction/sentence-classification model trained on our data? Note that unlike the previous pre-trained and naïve models, in this case we will need a 5-fold cross validation and compute corresponding mean and std for the various performance metrics
- [ ] Summarize all results in a histogram like the following:
    <img width="300" alt="Screen Shot 2022-08-16 at 14 52 46" src="https://user-images.githubusercontent.com/17013890/184884180-cd67fc5a-0261-4aa5-9f50-74c007612b47.png">
.

**Comment by jankrepl** [comment]
Here are the first results. We used `StratifiedKFold` with `n_splits=5`. One potential bias of our custom model (`textclassifier`) is that we used the the validations sets to do early stopping.

![Screenshot 2022-09-06 at 15 52 48](https://user-images.githubusercontent.com/18519371/188654066-1c00e243-097e-412f-b3d6-cc125168ff84.png)
![Screenshot 2022-09-06 at 15 53 06](https://user-images.githubusercontent.com/18519371/188654069-aacdd29b-3552-4df8-a834-86b261d8d0cc.png)
![Screenshot 2022-09-06 at 15 56 19](https://user-images.githubusercontent.com/18519371/188654074-5324362b-8811-4f7c-addd-34bf91708b03.png)

**Comment by jankrepl** [comment]
See below results after adding a test set

![Screenshot 2022-09-12 at 12 13 33](https://user-images.githubusercontent.com/18519371/189629578-31e6ff03-997e-4d02-ae13-82397edb6272.png)
![Screenshot 2022-09-12 at 12 15 29](https://user-images.githubusercontent.com/18519371/189629714-48120ea7-b0c4-4058-b2f8-fe3412c62421.png)
![Screenshot 2022-09-12 at 12 15 37](https://user-images.githubusercontent.com/18519371/189629718-d8b4958a-ca53-466a-ad35-482e4433ed36.png)
